src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -105,32 +113,184 @@ private static ServiceBusEntityType ConvertToServiceBusEntityType(ServiceBusEnti
+            
+            _serviceProvider = serviceProvider;
+            _tokenCredential = tokenCredential;
+
+            EntityName = entityName;
+            SubscriptionName = subscriptionName;
+            ServiceBusEntity = serviceBusEntity;
+            Options = options;
+            FullyQualifiedNamespace = fullyQualifiedNamespace;
+        }
+
+        /// <summary>
+        /// Gets the name of the Azure Service Bus entity to process.
         /// </summary>
         /// <remarks>This is optional as the connection string can contain the entity name</remarks>
         public string EntityName { get; }
 
         /// <summary>
-        ///     Name of the subscription to process
+        /// Gets the name of the Azure Service Bus Topic subscription.
         /// </summary>
         /// <remarks>This is only applicable when using Azure Service Bus Topics</remarks>
         public string SubscriptionName { get; }
 
         /// <summary>
-        ///     Entity of the Service Bus.
+        /// Gets the type of the Azure Service Bus entity.
         /// </summary>
         public ServiceBusEntityType ServiceBusEntity { get; }
 
         /// <summary>
-        ///     Options that influence the behavior of the message pump
+        /// Gets the fully qualified namespace where the Azure Service Bus entity is located.
+        /// </summary>
+        public string FullyQualifiedNamespace { get; }
+        
+        /// <summary>
+        /// Gets the additional options that influence the behavior of the message pump.
         /// </summary>
         public AzureServiceBusMessagePumpConfiguration Options { get; internal set; }
 
         /// <summary>
-        ///     Gets the configured connection string
+        /// Gets the administration client that handles the management of the Azure Service Bus resource.
+        /// </summary>
+        internal async Task<ServiceBusAdministrationClient> GetServiceBusAdminClientAsync()
+        {
+            if (_tokenCredential is null)
+            {
+                string connectionString = await GetConnectionStringAsync();
+                var client = new ServiceBusAdministrationClient(connectionString);
+                
+                return client;
+            }
+            else
+            {
+                var client = new ServiceBusAdministrationClient(FullyQualifiedNamespace, _tokenCredential);
+                return client;
+            }
+        }
+
+        /// <summary>
+        /// Determines the path based on the provided settings where the Azure Service Bus entity is located.
+        /// </summary>
+        internal async Task<string> GetEntityPathAsync()
+        {
+            if (_tokenCredential is null)
+            {
+                string connectionString = await GetConnectionStringAsync();
+                var properties = ServiceBusConnectionStringProperties.Parse(connectionString);
```

Comment:
What if the entity path is not part of the connection string?

Reply:
Good point, thanks! Missed that one.